[ML] Add suitable scoring functions for Text embedding models (#2183)
elastic/stack-docs: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc

[discrete]
[[ml-nlp-model-ref-text-embedding]]
== Third party text embedding models

> Text Embedding models are designed to work with specific scoring functions

Reviewer: ```suggestion

Text Embedding models are designed to work with specific scoring functions
```

for calculating the similarity between the embeddings they produce. 
Examples of typical scoring functions are: `cosine`, dot product` and 
`euclidean distance` (also known as `l2_norm`).

> The embeddings produced by these models should be indexed in Elasticsearch using the

Author: ```suggestion
The embeddings produced by these models should be indexed in {es} using the
```

> [dense vector field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html)

Author: ```suggestion
{ref}/dense-vector.html[dense vector field type]
```

with an appropriate {ref}/dense-vector.html#dense-vector-params[similarity function]
chosen for the model. 

To find similar embeddings in {es} use the efficient {ref}/knn-search.html#approximate-knn[Approximate k-nearest neighbor (kNN)]
search API with a text embedding as the query vector. 

> Approximate KNN search uses the similarty function defined in 

Author: ```suggestion
Approximate kNN search uses the similarity function defined in 
```

> the dense vector field mapping is used to calulate the relevence.

Reviewer: ```suggestion
the dense vector field mapping is used to calculate the relevance.
```

[discrete]
[[ml-nlp-model-ref-text-embedding]]
== Third party text embedding models
Text Embedding models are designed to work with specific scoring functions
for calculating the similarity between the embeddings they produce. 
Examples of typical scoring functions are: `cosine`, dot product` and 
`euclidean distance` (also known as `l2_norm`).

> The embeddings produced by these models should be indexed in Elasticsearch using the
> [dense vector field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html)
> with an appropriate [similarity function](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html#dense-vector-params)
> chosen for the model. 

Reviewer: ```suggestion
The embeddings produced by these models should be indexed in Elasticsearch using the
[dense vector field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html)
with an appropriate {ref}/dense-vector.html#dense-vector-params[similarity function]
chosen for the model. 
```

> To find similar embeddings in Elasticsearch use the efficient Approximate k-nearest neighbor (kNN)
> search API with a text embedding as the query vector. 
> https://www.elastic.co/guide/en/elasticsearch/reference/master/knn-search.html#approximate-knn

Reviewer: ```suggestion
To find similar embeddings in {es} use the efficient {ref}/knn-search.html#approximate-knn[Approximate k-nearest neighbor (kNN)]
search API with a text embedding as the query vector. 
```
